packages/ubuntu_desktop_installer/lib/pages/installation_complete/version_extractor.dart

```diff
@@ -0,0 +1,31 @@
+import 'dart:io';
+
+class VersionExtractor {
+  String _dummy() {
+    return 'Ubuntu 20.04';
+  }
+
+  String getVersion({String path = '/cdrom/.disk/info'}) {
+    final fileOnCdrom = File(path);
+    if (fileOnCdrom.existsSync()) {
+      try {
+        final content = fileOnCdrom
+            .readAsLinesSync()
+            .firstWhere((line) => line.trim().isNotEmpty);
+
+        return extractVersion(content);
+      } catch (e) {
+        //TODO: configure logger?
+      }
+    }
+    return _dummy();
+  }
+
+  String extractVersion(String content) {
+    final chunks = content.split(' ');
+    if (chunks.length > 1) {
+      return '${chunks[0]} ${chunks[1]}';
+    }
```

Comment:
I'm not sure whether there's a specification for the contents of that `info` file, but it looks to me like the format is roughly:

    PRODUCTNAME VERSION "CODENAME" - OTHERINFO

and what we want is PRODUCTNAME and VERSION. I can imagine PRODUCTNAME may contain whitespaces (e.g. "Ubuntu MATE"), and so may VERSION (e.g. "20.04.2.0 LTS"), so splitting on whitespaces isn't going to handle all cases nicely.

I'd recommend a regexp that extracts those components and keeps only PRODUCTNAME and VERSION.

Reply:
I checked few isos and here are my findings:
Xubuntu 20.04.2.0 LTS "Focal Fossa" - Release amd64 (20210209.1)
Ubuntu 20.04.2.0 LTS "Focal Fossa" - Release amd64 (20210209.1)
Kubuntu 21.04 "Hirsute Hippo" - Release amd64 (20210420)
Kubuntu 20.04.2.0 LTS "Focal Fossa" - Release amd64 (20210209.1)
Ubuntu-MATE 21.04 "Hirsute Hippo" - Release amd64 (20210420)
Ubuntu-Budgie 21.04 "Hirsute Hippo" - Release amd64 (20210420)

seems like instead of some regex which would introduce more problems probably (insert xkcd comic here) it would be the best to just substring to first `"`